src/topology/algebra/ring.lean

```diff
@@ -472,3 +472,18 @@ def to_add_group_topology.order_embedding : order_embedding (ring_topology α)
   end }
 
 end ring_topology
+
+section absolute_value
+
+/-- Construct an absolute value on a division ring `T` from an absolute value on a semiring `R`
+and ring homomorphism `f : T →+* R` -/
+def absolute_value.comp {R S T : Type*} [division_ring T] [semiring R] [nontrivial R]
+  [ordered_semiring S] (v : absolute_value R S) (f : T →+* R)  :
```

Comment:
If you change this to
```suggestion
  [ordered_semiring S] (v : absolute_value R S) (f : T →+* R) (hf : injective f) :
```
Then you can relax the typeclass requirements to ring or semiring

Reply:
Yeah, I thought about that but I thought it would be harder to use then. But I guess you're right, I should make it as general as possible.